llama_index/indices/multi_modal/base.py

```diff
@@ -292,40 +272,24 @@ def _add_nodes_to_index(
                 self._docstore.add_documents(
                     [node_without_embedding], allow_update=True
                 )
-        else:
-            # NOTE: if the vector store keeps text,
-            # we only need to add image and index nodes
-            for node, new_id in zip(all_nodes, all_new_ids):
-                if isinstance(node, (ImageNode, IndexNode)):
-                    # NOTE: remove embedding from node to avoid duplication
-                    node_without_embedding = node.copy()
-                    node_without_embedding.embedding = None
-
-                    index_struct.add_node(node_without_embedding, text_id=new_id)
-                    self._docstore.add_documents(
-                        [node_without_embedding], allow_update=True
-                    )
 
     def delete_ref_doc(
         self, ref_doc_id: str, delete_from_docstore: bool = False, **delete_kwargs: Any
     ) -> None:
         """Delete a document and it's nodes by using ref_doc_id."""
         # delete from all vector stores
+
         for vector_store in self._storage_context.vector_stores.values():
             vector_store.delete(ref_doc_id)
 
-            # delete from index_struct only if needed
-            if not self._vector_store.stores_text or self._store_nodes_override:
+            if self._store_nodes_override or self._vector_store.stores_text:
                 ref_doc_info = self._docstore.get_ref_doc_info(ref_doc_id)
                 if ref_doc_info is not None:
                     for node_id in ref_doc_info.node_ids:
                         self._index_struct.delete(node_id)
                         self._vector_store.delete(node_id)
 
-            # delete from docstore only if needed
-            if (
-                not self._vector_store.stores_text or self._store_nodes_override
-            ) and delete_from_docstore:
-                self._docstore.delete_ref_doc(ref_doc_id, raise_error=False)
+        if delete_from_docstore:
+            self._docstore.delete_ref_doc(ref_doc_id, raise_error=False)
 
-            self._storage_context.index_store.add_index_struct(self._index_struct)
+        self._storage_context.index_store.add_index_struct(self._index_struct)
```

Comment:
is it a overwrite or append?

Reply:
It's an overwrite, since the ID didn't change, but the list of nodes in the index struct will have changed

Reply:
got it thanks!